backend/src/main/kotlin/org/loculus/backend/service/submission/UploadDatabaseService.kt

```diff
@@ -237,11 +253,13 @@ class UploadDatabaseService(
                 .where { uploadIdColumn eq uploadId }
                 .map { it[accessionColumn]!! }
 
-        accessionPreconditionValidator.validate {
-            thatAccessionsExist(accessions)
-                .andThatUserIsAllowedToEditSequenceEntries(authenticatedUser)
-                .andThatSequenceEntriesAreInStates(listOf(Status.APPROVED_FOR_RELEASE))
-                .andThatOrganismIs(organism)
+        accessions.processInDatabaseSafeChunks { chunk ->
+            accessionPreconditionValidator.validate {
+                thatAccessionsExist(chunk)
+                    .andThatUserIsAllowedToEditSequenceEntries(authenticatedUser)
+                    .andThatSequenceEntriesAreInStates(listOf(Status.APPROVED_FOR_RELEASE))
+                    .andThatOrganismIs(organism)
+            }
```

Comment:
As above, I think that it _would_ be ideal if we can push it down to make the (reusable) validation functions safe on their own. However, I can imagine that it is not trivial in this case: if it's not easily done, should we create a ticket to consider further refactoring in the future?

Reply:
This should be done: here are the combined changes since last review - https://github.com/loculus-project/loculus/pull/4813/files/59296b76ebf5007052d6dfad69069777adb48dd0..fee16b976df009cf069e4599025b98b83dc09503